src/utils/solidity/transactionFactory.ts

```diff
@@ -136,25 +164,58 @@ const transactionFactory: TransactionFactory = <Args, Result>(
   guard = defaultGuard,
   prepareArgs = defaultPrepareArgs,
   postProcess = defaultPostProcess,
-  defaultOptions?,
+  defaultOptions = {},
 ) => {
   const prepare: PrepareFunction<Args> = async (
     contractAddress,
     params,
-    options = defaultOptions,
+    optionsOrCallback = defaultOptions,
     environment: Environment = getGlobalEnvironment(),
   ) => {
     await guard(params, contractAddress, environment);
     const args = await prepareArgs(params, contractAddress, environment);
     const contractInstance = getContract(contract, contractAddress);
     const transaction = contractInstance.methods[name](...args);
+
     transaction.name = name;
     const prepared = await prepareTransaction(
       transaction,
-      options,
+      optionsOrCallback,
       environment,
     );
-    return { ...prepared, contract };
+    const options: Options =
```

Comment:
@fubhy I had to change a lot here: 1. Add in AMGU calc, 2. return a JSON serializable transaction to be able to pipe it through graphql to the frontend and back.